Return types

orklah opened the pull request:
Hi,

In this PR, I had to modify the code a little. Mainly adding return null; or replacing return; by return null; for methods returning types.

orklah commented:
Mmh Psalm emit an issue that seems like a false positive. I reverted my edits in this file but the issue remains... I can't even reproduce it locally

weirdan commented:
As `$inner_context` is a by-ref param of that method it does indeed look like false positive. Anyway, I have the same issue emitted in an unrelated PR, so it's not something caused by your changes.

weirdan commented:
@orklah a lot of `@return`s seems redundant after these changes. Do we have a reason to keep them there?

orklah commented:
I removed some of them,, but I kept some of them on purpose:
- When a parent documentation is different than child native return type, Psalm complains that the two are incoherent. Redefining the phpdoc in the child fixes the issue
- A lot of case I fixed where false|null which is more precise than ?bool in signature

orklah commented:
> As `$inner_context` is a by-ref param of that method it does indeed look like false positive. Anyway, I have the same issue emitted in an unrelated PR, so it's not something caused by your changes.

oh, yeah, I see that it's on master too. Thanks for noticing

orklah commented on src/Psalm/Type/Atomic.php:
```diff
@@ -575,7 +575,7 @@ public function toNamespacedString(
         int $php_major_version,
         int $php_minor_version
-    );
+    ): ?string;
```
Well spotted, Dropping the parent phpdoc is enough to make psalm stop emitting issues in childs when type don't match.

muglug commented:
Thanks, sorry for breaking master
